store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/operation/getelements/impl/ParquetElementRetriever.java

```diff
@@ -188,12 +196,15 @@ public Element next() throws NoSuchElementException {
                 e = queue.poll();
                 if (null != e) {
                     if (needsValidation) {
```

Comment:
All validation and filters should really be applied by the RetrieveElementsFromFile class so that the functionality is computed by the threads per file rather than by a single thread.